front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx

```diff
@@ -73,7 +73,30 @@ const Itinerary = () => {
 
   const inverseOD = () => {
     notifyRestrictionResetWarning();
-    const newPathSteps = [...pathSteps].reverse();
+
+    // Reverse start and end of margins, in prevision of reversing the list of path steps
+    const newMargins: (string | undefined)[] = [];
+    let prevMargin: string | undefined;
+    pathSteps.forEach((pathStep, index) => {
```

Comment:
I think we can `compact` here, the button isn't available if there is no origin/destination (the only ones who can be `null`). This way, you can avoid the other checks for that in the function.
```suggestion
    compact.pathSteps.forEach((pathStep, index) => {
```

Reply:
Sure, done!